app.go

```diff
@@ -220,8 +258,14 @@ func (app *App) provide(constructor interface{}) {
 		return
 	}
 	app.logger.PrintProvide(constructor)
+
+	if _, ok := constructor.(Option); ok {
+		app.optionErr = fmt.Errorf("cannot Provide fx.Option: %v", constructor)
```

Comment:
Instead of just saying "cannot Provide fx.Option: ..." we should go the extra mile and say something like:

```
fx.Option should be passed to fx.New directly, not through fx.Provide: %v
```